File: C++/242-valid-anagram.cpp
#include <vector>
#include <string>
#include <algorithm>

using namespace std;

class Solution {
public:
    bool isAnagram(string s, string t) {
        std::vector<int> hist (26, 0);
        for (char c : s) hist[c - 'a']++;
        for (char c : t) hist[c - 'a']--;
        std::vector<int>::iterator p;
        p = std::find_if_not(hist.begin(), hist.end(), [](int i){return i == 0;});
        if (p == hist.end()) return true;
        return false;
    }
};

File: JS/242-valid-anagram.js
// hash table

/**
 * @param {string} s
 * @param {string} t
 * @return {boolean}
 */
var isAnagram = function(s, t) {
    let hist = Array(26).fill(0);
    for (let c of s) hist[c.codePointAt() - 'a'.codePointAt()]++;
    for (let c of t) hist[c.codePointAt() - 'a'.codePointAt()]--;
    if(hist.findIndex(cnt => cnt != 0) == -1) return true;
    else return false;
};